File: app/src/main/java/com/lucasurbas/counter/ui/detail/di/DetailFragmentComponent.java
package com.lucasurbas.counter.ui.detail.di;

import dagger.Subcomponent;
import com.lucasurbas.counter.app.di.FragmentModule;
import com.lucasurbas.counter.app.di.scope.FragmentScope;
import com.lucasurbas.counter.ui.detail.DetailFragment;

@Subcomponent(modules = {
        FragmentModule.class, DetailFragmentModule.class
})
@FragmentScope
public interface DetailFragmentComponent {

    void inject(DetailFragment fragment);
}

File: app/src/main/java/com/lucasurbas/counter/ui/detail/di/DetailFragmentComponent.kt
package com.lucasurbas.counter.ui.detail.di

import com.lucasurbas.counter.app.di.FragmentModule
import com.lucasurbas.counter.app.di.scope.FragmentScope
import com.lucasurbas.counter.ui.detail.DetailFragment
import dagger.Subcomponent

@Subcomponent(modules = arrayOf(FragmentModule::class, DetailFragmentModule::class))
@FragmentScope
interface DetailFragmentComponent {

    fun inject(fragment: DetailFragment)
}
